cmd/outline-ss-server/config.go

```diff
@@ -22,71 +22,128 @@ import (
 )
 
 type ServiceConfig struct {
-	Listeners []ListenerConfig
-	Keys      []KeyConfig
-	Dialer    DialerConfig
+	Listeners []ListenerConfig `yaml:"listeners"`
+	Keys      []KeyConfig      `yaml:"keys"`
+	Dialer    DialerConfig     `yaml:"dialer"`
 }
 
 type ListenerType string
 
-const listenerTypeTCP ListenerType = "tcp"
+const (
+	listenerTypeTCP ListenerType = "tcp"
 
-const listenerTypeUDP ListenerType = "udp"
+	listenerTypeUDP             ListenerType = "udp"
+	listenerTypeWebsocketStream ListenerType = "websocket-stream"
+	listenerTypeWebsocketPacket ListenerType = "websocket-packet"
+)
+
+type WebServerConfig struct {
+	ID        string   `yaml:"id"`
+	Listeners []string `yaml:"listen"`
+}
 
 type ListenerConfig struct {
-	Type    ListenerType
-	Address string
+	Type      ListenerType `yaml:"type"`
+	Address   string       `yaml:"address,omitempty"`
+	WebServer string       `yaml:"web_server,omitempty"`
```

Comment:
I found it best to avoid tags, since that ties you to a specific implementation. You can't use this for JSON or other libraries. Better to just name the fields like you want.

In this case, you could use `Web_Server` to remove the tag.